azurerm/internal/services/kusto/kusto_eventhub_data_connection_resource.go

```diff
@@ -83,6 +83,16 @@ func resourceKustoEventHubDataConnection() *schema.Resource {
 				ValidateFunc: eventhubValidate.EventHubID,
 			},
 
+			"event_system_properties": {
```

Comment:
event seems very vague here? as it applies to the eventhub:
```suggestion
			"eventhub_system_properties": {
```

Reply:
It does apply to eventhub, but those are properties of a single event as far as I understand, not of eventhub on its own.
@derSchtefan since you originally requested this, can you chip in on semantics please? :)

Reply:
These appear to be (System) Events - so the naming `event_system_properties` would make more sense than `eventhub_system_properties` here: https://docs.microsoft.com/en-us/azure/data-explorer/ingest-data-event-hub-overview#event-system-properties-mapping

Reply:
@katbyte @favoretti EventHub and IoT Hub are based around Kafka, a similar open-source system. EventHub and IoT Hub both attach SYSTEM properties to their events, that Kafka does not know about. One of them is x-opt-publisher which in conjunction with SAS token locked publisher endpoints allow you to identify who published an event (which is a nightmare to identify and make secure otherwise)

The property should be named event_system_properties because: 
* Microsoft names them like this not only in the portal but also the json schema
* They are not specific to EventHub, but also include IoT Hub

The list of event_system_properties can be found here:
* For EventHub based ingestions: https://docs.microsoft.com/en-us/azure/data-explorer/ingest-data-event-hub-overview#event-system-properties-mapping
* For IoT hub based ingestions: https://docs.microsoft.com/en-us/azure/data-explorer/ingest-data-iot-hub-overview#system-properties

IoT and EventHub both are mapped to EventHub ingestion internally, and work the same, since IoT hub is just an EventHub with some extra features.

Reply:
See also @tombuildsstuff 's comment here: https://github.com/terraform-providers/terraform-provider-azurerm/pull/11006#discussion_r597616052